### PR TITLE
GH-94398: TaskGroup: Fail create_task() during shutdown

### PR DESCRIPTION
**Pull request by gvanrossum**
Once the task group is shutting down,
it should not be possible to create a new task.
Here "shutting down" means `self._aborting` is set,
indicating that at least one task has failed and we have
cancelled all others.

* Issue: gh-94398

### CODE REVIEW

**Comment by gvanrossum** [comment]
I could really use a core dev review here. @1st1 do you have time? The idea is pretty simple (see the issue, GH-94398).

**Comment by gvanrossum** [comment]
@ambv feel like reviewing this? Pablo thought it would make a nice addition to the (still stuck) beta.

**Comment by ambv** [comment]
Looking.

File: Lib/test/test_asyncio/test_taskgroups.py [hunk 1]
```diff
@@ -722,6 +720,27 @@ async def coro(val):
...
+    async def test_taskgroup_no_create_task_after_failure(self):
+        async def coro1():
+            await asyncio.sleep(0.001)
+            1 / 0
+        async def coro2(g):
+            try:
+                await asyncio.sleep(100)
+            except asyncio.CancelledError:
+                with self.assertRaises(RuntimeError):
+                    g.create_task(c1 := coro1())
```
Comment (1st1) [inline Lib/test/test_asyncio/test_taskgroups.py:732]: Wow, walrus here feels elegant!

**Review by 1st1 — APPROVED** [review]

File: Misc/NEWS.d/next/Library/2022-06-29-04-42-56.gh-issue-94398.YOq_bJ.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Once an ``asyncio.TaskGroup`` has started shutting down (i.e., at least one task has failed and the task group has started cancelling the remaining tasks), it should not be possible to add new tasks to the task group.
```
Comment (ambv) [inline Misc/NEWS.d/next/Library/2022-06-29-04-42-56.gh-issue-94398.YOq_bJ.rst]: Let's wait for GH-94456 so we can properly refer to `asyncio.TaskGroup` in the Blurb.

```suggestion
Once a :class:`asyncio.TaskGroup` has started shutting down (i.e., at least one task has failed and the task group has started cancelling the remaining tasks), it should not be possible to add new tasks to the task group.
```

**Comment by gvanrossum** [comment]
Thanks, let me see if I can add something to the now-existing TaskGroup docs about these semantics.

**Comment by gvanrossum** [comment]
Sorry for the git mess. I really just merged main (twice, apparently) and added one line to the documentation: https://github.com/python/cpython/pull/94400/commits/5b9817d12e26c68d6d1f4526ed9ba706c8698c67

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**